### PR TITLE
383 get connected subgraphs in an alchemicalnetwork

### PR DESCRIPTION
**Pull request by atravitz**
addresses #383

### CODE REVIEW

**Comment by atravitz** [comment]
I'm not convinced that `connected_subgraphs` is the clearest language here. Most technically accurate would be `connected_components`, but given that `components` has a different meaning in openfe-world than in network theory world, I don't like that either.

Maybe `subgraphs` really is the best option?

File: gufe/network.py [hunk 1]
```diff
@@ -10,6 +11,7 @@
 from .transformations import Transformation
 
 
+
```
Comment (IAlibay) [inline gufe/network.py:14]:
```suggestion
```
We really should just apply `black` to gufe.

File: gufe/network.py [hunk 2]
```diff
@@ -119,3 +121,18 @@ def to_graphml(self) -> str:
     def from_graphml(cls, str):
```
Comment (IAlibay) [inline gufe/network.py:121]:
```suggestion
    def from_graphml(cls, str) -> Self:
```

File: gufe/tests/test_alchemicalnetwork.py [hunk 3]
```diff
@@ -47,3 +47,22 @@ def test_connectivity(self, benzene_variants_star_map):
             else:
                 edges = alnet.graph.edges(node)
                 assert len(edges) == 0
+
+    def test_connected_subgraphs(self, benzene_variants_star_map):
```
Comment (IAlibay) [inline gufe/tests/test_alchemicalnetwork.py:51]: could you add a brief test that checks that this works for fully connected graph? i.e. it just returns itself

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of small things, otherwise lgtm.

File: gufe/tests/test_alchemicalnetwork.py (hunk 3 above)
Comment (atravitz) [reply gufe/tests/test_alchemicalnetwork.py]: yes good idea. I rearranged the fixtures a bit so we can access either one or both of the ligand and complex star networks for the benzene test case. I think this is the clearest way to do this without duplicating too much code.

**Pull request merged**